app/controllers/subscriptions_controller.rb

```diff
@@ -32,15 +32,17 @@ def update
 
     # have adjusted presentation logic to only show the button to upgrade the subscription if the
     # user themselves actually paid for the original premium plan
-    customer = Stripe::Customer.retrieve(current_user.stripe_customer_id)
-    subscription = customer.subscriptions.retrieve(customer.subscriptions.first.id)
-    subscription.plan = "premiummob"
-    subscription.save
-    current_user.subscription.plan = Plan.find_by third_party_identifier: 'premiummob'
-    current_user.save
+    
+    plan = Plan.find_by(third_party_identifier: 'premiummob')
+    if user.subscription.payment_source.is_a? PaymentSource::Stripe
+      upgrade_plan_through_stripe(current_user, plan)
+    else
+      redirect_to new_subscription_path(plan: plan.third_party_identifier)      
```

Comment:
I'm worried about redirect premiums to this page - there's a danger they sign up for Premium Mob and buy a second subscription - if we had something to tell them it was there responsibility to cancel their original Paypal account - and also CA grads would end up here ... but this is better than nothing I guess - maybe we need to make handling that part of another ticket and get this merged ... if it happens we can quickly get in there to request they shut down the original paypal subscription, or give CA's a discount etc ...